sklearn/svm/_classes.py

```diff
@@ -271,6 +277,25 @@ def fit(self, X, y, sample_weight=None):
         check_classification_targets(y)
         self.classes_ = np.unique(y)
 
+        if self.dual == "auto":
+            if self.loss == "hinge":
+                self._dual = True
+            elif self.loss == "squared_hinge" and self.penalty == "l1":
+                self._dual = False
+            else:
+                self._dual = True if X.shape[0] < X.shape[1] else False
```

Comment:
I think that we could avoid redefining the logic here and reuse the `_get_liblinear_solver_type` function.

```suggestion
        if self.dual == "auto":
            if X.shape[0] < X.shape[1]:
                try:
                    # check if the combination of loss and penalty is supporting
                    # dual formulation
                    _get_liblinear_solver_type(
                        self.multi_class, self.penalty, self.loss, True
                    )
                    self._dual = True
                except ValueError:
                    self._dual = False
            else:
                self._dual = False
```

Reply:
I had to reformulate this algorithm a bit, but I think it's working now